src/core/thread/mle.cpp

```diff
@@ -2305,6 +2305,7 @@ Error Mle::SendChildUpdateRequest(void)
     if (!IsRxOnWhenIdle())
     {
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+        Get<DataPollSender>().RecalculatePollPeriod();
```

Comment:
What scenario requires adding this here?

Reply:
When user set a new CSL timeout period to SSED, SSED will schedule a Child Update Request. Ref: https://github.com/openthread/openthread/blob/main/src/core/mac/mac.cpp#L2533

Reply:
For Child Timeout, we make this call early in `Mle::SetTimeout()` - see https://github.com/openthread/openthread/blob/21417407dd4b446b4775d889948e9eb06852da82/src/core/thread/mle.cpp#L727

Should we be more consistent in when we call `RecalculatePollPeriod()`? For example, should we move the call in `Mle::SetTimeout()` to here? Or should we move this call into `Mle::SetCslTimeout()`?

Reply:
SGTM, updated.